.github/workflows/static_analysis.yml

```
@@ -0,0 +1,52 @@
+# contains tests that can run without compiling LightGBM, like
+# linters and documentation checks.
+name: Static Analysis
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+env:
+  CONDA_ENV: test-env
+  GITHUB_ACTIONS: 'true'
+  PYTHON_VERSION: 3.8
+
+jobs:
+  test:
+    name: ${{ matrix.task }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - task: lint
+          - task: check-docs
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 5
+          submodules: true
```

To speedup. I believe empty submodule folders will not break anything.
```suggestion
          submodules: false
```

---

oooo yeah you're correct. I was thinking about the problem we had in https://github.com/microsoft/LightGBM/issues/3655, but that was only for the R package because the R package has to be installed for its site to be built.